apps/blockchain/lib/blockchain/mining.ex

```diff
@@ -73,7 +73,7 @@ defmodule Blockchain.Mining do
   end
 
   defp mine_block(%Block{} = b) do
-    mined_block = Block.perform_proof_of_work(b)
+    mined_block = ProofOfWork.compute(b)
```

Comment:
Wouldn't be better to make this a config that you could swap? If not then this implementation `force` you to use PoW which it is fine but in my head is probably better to create some config, even we can plug more than one computation (but that's too fancy for now)